### PR TITLE
refactor(app): pUR FF Removal: Nav

### LINKED ISSUE #8873: PUR FF Removal: Nav
All instances of `preProtocolFlowWithoutRPC` and their logic should be removed from the following locations within the nav.

```
opentrons/app/src/App/hooks.ts:
  40  export function useNavLocations(): NavLocation[] {
  41:   const isPreProtocolFlowWithoutRPC = useFeatureFlag(
  42:     'preProtocolFlowWithoutRPC'
  43    )

  52  
  53:   return isPreProtocolFlowWithoutRPC ? navLocations : legacyLocations
  54  }

opentrons/app/src/App/__tests__/hooks.test.tsx:
  124        when(mockUseFeatureFlag)
  125:         .calledWith('preProtocolFlowWithoutRPC')
  126          .mockReturnValue(true)

  140        when(mockUseFeatureFlag)
  141:         .calledWith('preProtocolFlowWithoutRPC')
  142          .mockReturnValue(false)

opentrons/app/src/redux/nav/__tests__/selectors.test.ts:
  131        enableBundleUpload: false,
  132:       preProtocolFlowWithoutRPC: false,
  133      })
```

### PR DESCRIPTION
**Pull request by smb2268**
close #8873

# Overview
Remove FF from Navigation

# Changelog
All instances of `preProtocolFlowWithoutRPC` and their logic should be removed from the following locations within the nav.

- [ ] opentrons/app/src/App/hooks.ts
- [ ] opentrons/app/src/App/__tests__/hooks.test.tsx
- [ ] opentrons/app/src/redux/nav/__tests__/selectors.test.ts

# Review requests
Verify that the new nav loads whether FF is on or off

# Risk assessment
low

### CODE REVIEW

File: app/src/App/hooks.ts [hunk 1]
```diff
@@ -38,17 +37,11 @@ export function useRunLocation(): NavLocation {
 }
 
 export function useNavLocations(): NavLocation[] {
-  const isPreProtocolFlowWithoutRPC = useFeatureFlag(
-    'preProtocolFlowWithoutRPC'
-  )
-  const [robots, upload, calibrate, legacyRun, more] = useSelector(
-    getNavbarLocations
-  )
+  const [robots, upload, more] = useSelector(getNavbarLocations)
```
Comment (b-cooper) [inline app/src/App/hooks.ts:40]: This is not behaving as you intended.  Be careful when destructuring arrays, the indices will be maintained even if the names change locally. In the current implementation `more` is actually the contents of the calibration location (which is the 2 index returned from getNavbarLocations).  You will either need to edit the contents of the `getNavbarLocations` selector to only return `robots`, `upload`, and `more`, or change this line to 

```
const [robots, upload, _legacyCalibrate, _legacyRun, more] = useSelector(
  getNavbarLocations
)
```

**Review by b-cooper — CHANGES_REQUESTED** [review]
Mostly good, just one bug that needs to be addressed

File: app/src/App/hooks.ts (hunk 1 above)
Comment (smb2268) [reply app/src/App/hooks.ts:40]: Ah my mistake - I'll push the former fix and make a TODO to clean up the `nav/__tests__/selectors.test.tsx` file for when we're ripping out more of the unnecessary code next week.

**Review by jerader — APPROVED** [review]
did a quick search and didn't find any instances of `preProtocolFlowWithoutRPC` 🚀

**Review by b-cooper — APPROVED** [review]
💣

**Pull request merged**